### PR TITLE
structured logging

### PR DESCRIPTION
**Pull request by Hades32**
This changes the runtime to use structured logging, just like https://github.com/product-os/transformer-runtime/pull/40 .

I also introduce a new linter rule to ensure nobody is using non-structured logging accidentally.

This should finally allow us to filter logs by PR (actually per commit).

When reviewing this, it makes sense to look at the individual commits, as some reformatting happened, too, because the linter wasn't enforced previously.

### CODE REVIEW

**Comment by Hades32** [comment]
Built successfully https://dashboard.balena-cloud.com/fleets/1780230/releases/2041032

File: transformer-runner/src/runner.ts [hunk 1]
```diff
@@ -40,7 +42,7 @@ const runtime = new Runtime(
 );
 
 export async function initializeRunner() {
-	console.log(`[WORKER] starting...`);
+	logger.info(`starting...`);
```
Comment (stathismor) [inline transformer-runner/src/runner.ts:45]: Minor: `starting` -> `Starting`. I am not sure there is a convention though, noticed this elsewhere.

File: transformer-runner/src/runner.ts [hunk 2]
```diff
@@ -49,60 +51,68 @@ export async function initializeRunner() {
...
-		try {
-			await produceErrorContract(task, err);
-		} catch (err: any) {
-			console.log(`[WORKER] ERROR couldn't set task to failed`, err);
-		}
-	} finally {
-		runningTasks.delete(task.id!);
-		await locker.removeActive();
-	}
+	await withLogger(
```
Comment (stathismor) [inline transformer-runner/src/runner.ts:75]: I think I get what this does, but why is it needed here?

File: transformer-runner/src/runner.ts [hunk 3]
```diff
@@ -49,60 +51,68 @@ export async function initializeRunner() {
...
-	console.log('[WORKER] Waiting for tasks');
+	logger.info('Waiting for tasks');
 }
 
 const acceptTask = async (update: { data?: { after?: TaskContract } }) => {
 	const task = update?.data?.after;
 	const taskStartTimestamp = Date.now();
 	if (!task) {
-		console.log(`[WORKER] WARN got an empty task`);
+		logger.info(`WARN got an empty task`);
```
Comment (stathismor) [inline transformer-runner/src/runner.ts:68]: Isn't this a `logger.warn`?

**Review by stathismor — COMMENTED** [review]
LGTM!

**Comment by Hades32** [comment]
@balena-ci I self-certify!

**Pull request merged**